File: code_examples/5-Holding_Data/2-Lists/Python/main.py
#!/usr/bin/env python3
"""
Created by: Mr. Coxall
Created on: Sep 2020
This module program uses a list
"""


def main() -> None:
    """The main() this function uses a list, returns None."""

    # input
    words = []
    temp_word = "Go"

    # input
    print("Please enter 1 word at a time. Enter STOP to end.")

    while temp_word.upper() != "STOP":
        temp_word = input("Enter a word: ")
        words.append(temp_word)

    words.pop()  # remove the "Stop" that was added
    print("")

    print("Here are the words reversed:")
    for _ in range(0, len(words)):
        print(words.pop())


    print("\nDone.")


if __name__ == "__main__":
    main()

File: code_examples/5-Holding_Data/2-Lists/JavaScript/main.js
/* Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program uses a list 
 */

const prompt = require("prompt-sync")()

// input
const words = [];
let tempWord = 'Go';

console.log('Please enter 1 word at a time. Enter STOP to end.');

while (tempWord.toUpperCase() !== 'STOP') {
  tempWord = prompt('Enter a word: ');
  words.push(tempWord);
}

words.pop(); // remove the "STOP" that was added
console.log();

console.log('Here are the words reversed:');
for (let loopCounter = words.length - 1; loopCounter >= 0; loopCounter--) {
  console.log(words[loopCounter]);
}

console.log("\nDone.")
